App/Sagas/ImageSharingSagas.ts

```diff
@@ -0,0 +1,101 @@
+import { call, put, select } from 'redux-saga/effects'
+import RNFS from 'react-native-fs'
+import uuid from 'uuid/v4'
+
+import { uploadFile } from './UploadFile'
+import getDefaultThread from './GetDefaultThread'
+import TextileNode from '../../TextileNode'
+import { SharedImage, AddResult, Thread } from '../Models/TextileTypes'
+import TextileNodeActions from '../Redux/TextileNodeRedux'
+import ProcessingImagesActions, { ProcessingImage, ProcessingImagesSelectors } from '../Redux/ProcessingImagesRedux'
+import UIActions from '../Redux/UIRedux'
+
+export function * shareWalletImage (id: string, threadId: string, comment?: string) {
+  try {
+    // TODO: Insert some state into the processing photos redux in case this takes long or fails
+    const blockId: string = yield call(TextileNode.sharePhotoToThread, id, threadId, comment)
+    yield put(TextileNodeActions.getPhotoHashesRequest(threadId))
+  } catch (error) {
+    yield put(UIActions.imageSharingError(error))
+  }
+}
+
+export function * insertImage (image: SharedImage, threadId: string, comment?: string) {
+  const id = uuid()
+  yield put(ProcessingImagesActions.insertImage(id, image, threadId, comment))
+  yield call(addToIpfs, id)
+}
+
+export function * addToIpfs (uuid: string) {
```

Comment:
Idea here is that all these sagas should be able to be called repeatedly (for retry), and the redux state will be consistent and the sequence of things that need to happen to share a photo will continue no matter where you start the retry from in the sequence.

Reply:
cool